### PR TITLE
AAP-32019: Enable Playbook generation/explanation endpoint in Lightspeed service for on-prem

### PR DESCRIPTION
**Pull request by manstis**
Jira Issue: https://issues.redhat.com/browse/AAP-32019

## Description
This PR adds support for Playbook Generation and Explanation for "on prem".

It is enabled/disabled with the environment variable `ENABLE_PLAYBOOK_ENDPOINT`

## Testing
TBD.

Once the container has been built by this PR I'll deploy an AAIC instance "on prem", test and document.

### Steps to test
1. Pull down the PR
2. Once the container has been built by this PR I'll deploy an AAIC instance "on prem", test and document.

### Scenarios tested
As above.

## Production deployment
- [ ] This code change is ready for production on its own
- [x] This code change requires the following considerations before going to production:

This PR depends on https://github.com/ansible/ansible-wisdom-ops/pull/1096

### CODE REVIEW

File: ansible_ai_connect/ai/api/model_pipelines/wca/pipelines_base.py [hunk 1]
```diff
@@ -389,10 +394,132 @@ class WCABasePlaybookGenerationPipeline(
     def __init__(self, inference_url):
         super().__init__(inference_url=inference_url)
 
+    def invoke(self, params: PlaybookGenerationParameters) -> PlaybookGenerationResponse:
```
Comment (manstis) [inline ansible_ai_connect/ai/api/model_pipelines/wca/pipelines_base.py:397]: I just moved this function from "SaaS" to the common base class for both "SaaS" and "on-prem".

File: ansible_ai_connect/ai/api/model_pipelines/wca/pipelines_base.py [hunk 2]
```diff
@@ -389,10 +394,132 @@ class WCABasePlaybookGenerationPipeline(
...
+
 
 class WCABasePlaybookExplanationPipeline(
     WCABasePipeline, ModelPipelinePlaybookExplanation, metaclass=ABCMeta
 ):
 
     def __init__(self, inference_url):
         super().__init__(inference_url=inference_url)
+
+    def invoke(self, params: PlaybookExplanationParameters) -> PlaybookExplanationResponse:
```
Comment (manstis) [inline ansible_ai_connect/ai/api/model_pipelines/wca/pipelines_base.py:474]: I just moved this function from "SaaS" to the common base class for both "SaaS" and "on-prem".

File: ansible_ai_connect/main/settings/base.py [hunk 3]
```diff
@@ -612,3 +612,9 @@ def is_ssl_enabled(value: str) -> bool:
 CHATBOT_DEFAULT_PROVIDER = os.getenv("CHATBOT_DEFAULT_PROVIDER")
 CHATBOT_DEFAULT_MODEL = os.getenv("CHATBOT_DEFAULT_MODEL")
 # ==========================================
+
+# =============================================
+# Playbook Generation/Explanation for "on-prem"
+# ---------------------------------------------
+ENABLE_PLAYBOOK_ENDPOINT = os.getenv("ENABLE_PLAYBOOK_ENDPOINT", "False").lower() == "true"
+# =============================================
```
Comment (hasys) [inline ansible_ai_connect/main/settings/base.py:620]: Looks like default version is `False`, it means PR should have a note in **Production deployment** section of this PR

Comment (manstis) [reply ansible_ai_connect/main/settings/base.py:620]: Ah, well, @hasys at the moment I only check the flag for "on prem".. so the endpoints are *always* enabled for SaaS.

I asked @jameswnl to clarify the JIRA Acceptance Criteria.. easy enough to use for both SaaS and "on prem".

See `#team-wisdom-engineering`.

Comment (manstis) [reply ansible_ai_connect/main/settings/base.py]: @hasys @jameswnl I've changed the PR to use the new environment variable for both SaaS and "on prem".

There is a corresponding [PR](https://github.com/ansible/ansible-wisdom-ops/pull/1096) for `ansible-wisdom-ops` to enable it for SaaS.

Local development (using `development.py` settings) also [enables](https://github.com/ansible/ansible-ai-connect-service/pull/1375/files#diff-30c4274265178ee8e8dfb34bd09feacd9d166040ee15fe3d9be42cf923996897R75) it by default.

**Pull request closed without merging**